controllers/tiertemplateguard/tiertemplate_guard_controller.go

```diff
@@ -0,0 +1,104 @@
+package tiertemplateguard
```

Comment:
Let's keep the naming consistency with the rest of the controllers - this is a standard controller that watches TierTemplates as it's primary resource. In the future, it could execute some validations and update the status of the CRs.
For now it only takes care of the finalizer, but managing the finalizer is always the role of the "main" resource controller.
```suggestion
package tiertemplate
```